### PR TITLE
Prevent users from misusing _PIP_ADDITIONAL_REQUIREMENTS

### PR DESCRIPTION
**Pull request by potiuk**
This PR aims to prevent a misuse of _PIP_ADDITIONAL_REQUIREMENTS by mistakes that our users often make without realising the consequences.

1) The _PIP_ADDITIONAL_REQUIREMENTS should only be used for quick
   iterations and testing, they should never be used in anything
   resembling production environment. This PR will give maximum
   10 minutes of life for a container that has the environment
   variable set. Should be enough for testing but unusable for
   production.

2) When installing the _PIP_ADDITIONAL_REQUIREMENTS we additionally
   install airflow version corresponding to the airflow version
   already installed in the image. This prevents accidental
   downgrades or upgrades of airflow by installing dependencies that
   are not compatible with given airflow version

3) After installing the depdencies with _PIP_ADDITIONAL_REQUIREMENTS
   we run `pip check` to make sure that newly installed requirements
   are not in conflict with some other requirements installed earlier.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/docker-stack/changelog.rst [hunk 1]
```diff
@@ -58,6 +58,21 @@ here so that users affected can find the reason for the changes.
 Airflow 2.7
 ~~~~~~~~~~~
 
+* 2.7.2
+
+  * Several fixes to how ``__PIP_ADDITIONAL_REQUIREMENTS`` works:
```
Comment (Taragolis) [inline docs/docker-stack/changelog.rst]:
```suggestion
  * Several fixes to how ``_PIP_ADDITIONAL_REQUIREMENTS`` works:
```

File: docs/docker-stack/changelog.rst [hunk 2]
```diff
@@ -58,6 +58,21 @@ here so that users affected can find the reason for the changes.
...
+
+    * When installing additional requirements, ``airflow_version=<AIRFLOW_VERSION>`` is passed to the pip command. This ensures that the
+      airflow version is never accidentally downgraded or upgraded when installing incompatible requirements.
+
+    * Airflow runs ``pip check`` after installing the requirements which will fail if ``pip`` encounters incompatible dependencies after installation -
+      this might happen in case some earlier installed dependencies are in conflict with the newly installed ones. Failure of pip check will fail
+      the image.
+
+    * Setting __PIP_ADDITIONAL_REQUIREMENTS will cause the image to shutdown after 10 minutes. This has always been a development only feature
```
Comment (Taragolis) [inline docs/docker-stack/changelog.rst]:
```suggestion
    * Setting _PIP_ADDITIONAL_REQUIREMENTS will cause the image to shutdown after 10 minutes. This has always been a development only feature
```

File: scripts/docker/entrypoint_prod.sh [hunk 3]
```diff
@@ -296,10 +296,17 @@ if [[ -n "${_AIRFLOW_WWW_USER_CREATE=}" ]] ; then
...
+    COLOR_RESET=$'\e[0m'
+    COLOR_YELLOW=$'\e[33m'
+
+    # The container will shutdown after that many seconds of use automatically when _PIP_ADDITIONAL_REQUIREMENTS ARE USED
+    KILL_TIMEOUT=600
+    >&2 echo
+    >&2 echo "${COLOR_YELLOW}Installing additional requirements: '${_PIP_ADDITIONAL_REQUIREMENTS}' ${COLOR_RESET}"
     >&2 echo
-    >&2 echo "!!!!!  Installing additional requirements: '${_PIP_ADDITIONAL_REQUIREMENTS}' !!!!!!!!!!!!"
+    >&2 echo "${COLOR_YELLOW}WARNING: This is a development/test feature only. NEVER use it in production!${COLOR_RESET}"
```
Comment (Taragolis) [inline scripts/docker/entrypoint_prod.sh:308]: I'm not sure about **development/test feature only**, I would never recommend to use it in development (deploy into dev environment) or test (deploy into the test/qa/preproduction environment).

I thought how it propagated into the prod: It work in local development, let's deploy on dev/test/staging

Comment (potiuk) [reply scripts/docker/entrypoint_prod.sh:308]: I am writing a message about this PR to devlist. Hold your horses @Taragolis  :D

**Pull request closed without merging**